compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1749,10 +1751,51 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
         }
 
+        let mut sugg_paths = vec![];
```

Comment:
`report_privacy_error` is more than 200 lines long after this change ^^ and i find it really difficult to read because all the path collection logic is all over the place. I think it would be better if we would re-structure some of the logic into functions. But it's not too important to me since this is only diagnostics code, so up to you whether you want to change this or not.

Reply:
Let's split this method in a follow up PR.